src/fileNamer.js

```diff
@@ -151,3 +151,139 @@ function mapValue(map, value) {
     return `-${map[value]}`
   }
 }
+
+function evaluateNewLabel(documents, L, LWP) {
+  let failed = 0
+  for (let doc of documents) {
+    doc.category = evalCategory(doc)
+    if (doc.category === undefined) {
+      if (LWP) {
+        log('info', `New label: ${doc.label}`)
+      }
+      log('warn', 'Category Unknown')
+      failed++
+      continue
+      //throw 'Category Unknown'
+    }
+    doc.type = evalType(doc)
+    if (doc.type === undefined) {
+      if (LWP) {
+        log('info', `New label: ${doc.label}`)
+      }
+      log('warn', 'Type Unknown')
+      failed++
+      continue
+      //throw 'Type Unknown'
+    }
+    doc.form = evalForm(doc, doc.year, doc.category)
+    if (doc.form === undefined) {
+      if (LWP) {
+        log('info', `New label: ${doc.label}`)
+      }
+      log('warn', 'Form Unknown')
+      failed++
+      continue
+      //throw 'Form Unknown'
+    }
```

Comment:
We have the same form of code three times. I suggest to refactor such as the relationship between the attributes and the functions trying to deduce what is the attribute is clearer.
    
```
const attributesToDeducer = {
  form: evalForm, // I would rename it to deduceForm
  type: evalType,
  category: evalCategory
}

for (let attr of attributesToDeducer) {
  const deducer = attributesToDeducer[attr]
  doc[attr] = deducer(doc)
    if (doc[attr] === undefined) {
      if (LWP) {
        log('info', `New label: ${doc.label}`)
      }
      log('warn', `${attr} could not be deduced`)
      failed++
      continue
    }
}
```